lib/layers/data/local/data_sources/personal_data_local_source.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:wot_statistic/layers/data/sources/local/local_data_source.dart';
+import 'package:wot_statistic/layers/data/models/local/user_data.dart';
+
+import 'drift_database/dao/wot_stat_dao.dart';
+
+const signedUserId = 'Singed User id';
+const signedUserNickname = 'Singed User nickname';
+const signedUserToken = 'Singed User token';
+const signedUserExpire = 'Singed User EXPIRE';
+const signedUserRealm = 'Singed User realm';
+const notPicked = "Not Picked";
+const realmKey = 'Realm';
+
+class PersonalDataLocalSourceImpl extends PersonalDataLocalSource {
+  final SharedPreferences sharedPreferences;
+  final WotStatDao wotStatDao;
+
+  PersonalDataLocalSourceImpl({
+    required this.sharedPreferences,
+    required this.wotStatDao,
+  });
+
+  @override
+  UserData? getSignedUser() {
```

Comment:
i see 2 simple solutions:
- from di module get from SP UserData and inject it instance into constructor.
- create a separate data source which returns UserData 
I prefer the second solution to avoid additional code in Di file

Reply:
after some time of thinking and reading comments, i decided to split local data sources (based on SharedPreferences) by sense, such as : LanguageSettings, UserRep e.t.